tests/utils.py

```diff
@@ -19,81 +19,95 @@
+            bbox,
+        )
+        morph = scl.models.parametric.sersic((n,), frame).astype(np.float32)
+        self.morph = morph/np.max(morph)
 
-    _seds = [
-        np.arange(B, dtype=float),
-        np.arange(B, dtype=float)[::-1],
-        np.ones((B,), dtype=float),
-    ]
-    seds = np.array([_seds[n % 3] * amplitudes[n] for n in range(K)], dtype=np.float32)
 
-    morphs = np.zeros((K, Ny, Nx), dtype=np.float32)
-    for k, coord in enumerate(coords):
-        morphs[k, coord[0], coord[1]] = 1
-    images = seds.T.dot(morphs.reshape(K, -1)).reshape(shape)
+class PsfModel(DeblenderTestModel):
+    def __init__(
+        self,
+        center,
+        spectrum: np.ndarray,
+        bands: Sequence[str],
+        radius: int = 7,
+    ):
+        self.center = center
+        self.spectrum = spectrum.astype(np.float32)
+        self.bands = bands
+        bbox = scl.Box((2*radius+1, 2*radius+1), (center[0]-radius, center[1]-radius))
+        self.bbox = bbox
+        self.morph = np.zeros(bbox.shape, dtype=np.float32)
+        _center = (self.morph.shape[0]-1)//2, (self.morph.shape[1]-1)//2
+        self.morph[*_center] = 1
 
-    if convolve:
-        psfRadius = 20
-        psfShape = (2 * psfRadius + 1, 2 * psfRadius + 1)
-        x = np.arange(-psfRadius, psfRadius, psfShape[0])
-        y = x.copy()
 
-        targetPsfImage = scl.utils.integrated_circular_gaussian(sigma=0.9, x=x, y=y)
+def initData(
+    models: list[DeblenderTestModel],
+    modelPsf: np.ndarray,
+    imagePsf: np.ndarray,
+) -> tuple[scl.Image, scl.Image]:
+    # Find the bounding box that contains all of the models
+    bbox = models[0].bbox
+    bands = models[0].bands
+    for model in models[1:]:
+        bbox = bbox | model.bbox
+        assert bands == model.bands
+    bbox = bbox.grow(5)
 
-        psfs = [GaussianPsf(psfShape[1], psfShape[0], 1 + 0.2 * b) for b in range(B)]
-        psfImages = np.array(
-            [psf.computeImage(psf.getAveragePosition()).array for psf in psfs]
-        )
-        psfImages /= psfImages.max(axis=(1, 2))[:, None, None]
-
-        # Convolve the image with the psf in each channel
-        # Use scipy.signal.convolve without using FFTs as a sanity check
-        images = np.array(
-            [
-                scipy.signal.convolve(img, psf, method="direct", mode="same")
-                for img, psf in zip(images, psfImages)
-            ]
-        )
-        # Convolve the true morphology with the target PSF,
-        # also using scipy.signal.convolve as a sanity check
-        morphs = np.array(
-            [
-                scipy.signal.convolve(m, targetPsfImage, method="direct", mode="same")
-                for m in morphs
-            ]
-        )
-        morphs /= morphs.max()
-        psfImages /= psfImages.sum(axis=(1, 2))[:, None, None]
+    for dim in bbox.origin:
+        if dim < 0:
+            raise ValueError("Invalid setup, at least one source has a footprint below (0, 0)")
```

Comment:
Why only check for the origin being below `(0, 0)` rather than all corners being within the scene?